### PR TITLE
[Dashboard] Postgrest API settings renamed to "Data API" settings. now using react-hook-form

### PR DESCRIPTION
**Pull request by MildTomato**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

- refactor API postgrest settings to react-hook-form
- schema selection stops submitting on change now 
- be aware, types have been changed

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 1]
```diff
@@ -98,163 +129,213 @@ const PostgrestConfig = () => {
...
+                      label="Max rows"
+                      description="The maximum number of rows returned from a view, table, or stored procedure. Limits payload size for accidental or malicious requests."
+                    >
+                      <FormControl_Shadcn_>
+                        <Input_Shadcn_
+                          size="small"
+                          disabled={!canUpdatePostgrestConfig}
+                          {...field}
+                          type="number"
+                          {...form.register('maxRows', {
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:296]: wondering why this one has valueAsNumber: true and the one below doesn't. 
just going to check

**Comment by saltcod** [comment]
looks good & works!  love the new multiselect 

This does suffer from a similar problem to before — you make the change at the top of this panel in the Multiselect, and the save/cancel buttons are way down below.  It isn't super obvious that something has changed and needs saving. 


screenshot-2024-05-28-at-21 55 46

we could move the actions to the top — only showing them when something's changed 
screenshot-2024-05-28-at-21 55 26

could also maybe do something like this — a tooltip trigger that says "Unsaved changes" or something 
screenshot-2024-05-28-at-22 00 00
.
Don't think this is a huge problem — but I bet this is why it was autosave in the first place

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 2]
```diff
@@ -41,7 +55,13 @@ const PostgrestConfig = () => {
...
   })
 
-  const { data: config, isError } = useProjectPostgrestConfigQuery({ projectRef })
+  const {
+    data: config,
+    isLoading: isLoadingPostgrestConfigQuery,
+    isError,
+  } = useProjectPostgrestConfigQuery({ projectRef })
+
+  console.log('config', config)
```
Comment (saltcod) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:64]: ^

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 3]
```diff
@@ -98,163 +129,213 @@ const PostgrestConfig = () => {
...
+        dbSchema: config.db_schema ? config.db_schema.replace(/ /g, '').split(',') : [],
+        maxRows: config.max_rows,
+        dbExtraSearchPath: config.db_extra_search_path,
+        dbPool: config.db_pool,
+      })
+    }
+  }
 
-            resetForm({ values, initialValues: values })
+  useEffect(() => {
```
Comment (saltcod) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:143]: can you use the isSuccess callback here like Ivan suggested for that other useEffect issue?

**Review by saltcod — APPROVED** [review]

**Review by joshenlim — CHANGES_REQUESTED** [review]
there seems to be some height style issue going on in the settings/api page here
Notice the empty space at the bottom

It seems like the MultiSelector component is the one causing it actually - i tried commenting out line 192 specifically and its okay

image

**Comment by MildTomato** [comment]
> there seems to be some height style issue going on in the settings/api page here
Notice the empty space at the bottom

I can't repro this. anything in particular you did to repro ?

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 2 above)
Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx]: thanks! fixed

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 3 above)
Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:141]: Can we change this in a follow up PR. Didn't want to get involved in changing how the form resets in this one.

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 4]
```diff
@@ -98,163 +127,213 @@ const PostgrestConfig = () => {
...
+                          <MultiSelectorTrigger>
+                            <MultiSelectorInput placeholder="Select schemas for Data API..." />
+                          </MultiSelectorTrigger>
+                          <MultiSelectorContent>
+                            {schema.length <= 0 ? (
+                              <MultiSelectorList>
+                                <MultiSelectorItem key={'empty'} value={'no'}>
+                                  no
+                                </MultiSelectorItem>
+                              </MultiSelectorList>
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:205]: 😅 ah, that needs to go

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:205]: i'll remove this

**Review by joshenlim — APPROVED** [review]

**Pull request merged**